packages/renderer/src/stores/kubernetes-contexts.ts

```diff
@@ -0,0 +1,61 @@
+/**********************************************************************
+ * Copyright (C) 2023 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import { writable, type Writable } from 'svelte/store';
+import { EventStore } from './event-store';
+import { getKubeUIContexts, type KubeContextUI } from '../lib/kube/KubeContextUI';
+
+const windowEvents: string[] = [
+  'extensions-started',
+  'kubernetes-context-update',
+  'kubernetes-context-create',
+  'kubernetes-context-delete',
+];
+const windowListeners = ['extensions-already-started'];
+
+// Do not update until all extensions are started (since we want to make sure kube-context has been loaded)
+let readyToUpdate = false;
+export async function checkForUpdate(eventName: string): Promise<boolean> {
+  if ('extensions-already-started' === eventName) {
+    readyToUpdate = true;
+  }
+
+  // do not fetch until extensions are all started
+  return readyToUpdate;
+}
+
+export const kubernetesContexts: Writable<KubeContextUI[]> = writable([]);
+
+const eventStore = new EventStore<KubeContextUI[]>(
+  'kubernetes contexts',
+  kubernetesContexts,
+  checkForUpdate,
+  windowEvents,
+  windowListeners,
+  grabKubernetesContexts,
+);
+eventStore.setup();
+
+export async function grabKubernetesContexts(): Promise<KubeContextUI[]> {
+  // Retrieve both the contexts and clusters
+  const k8sContexts = await window.kubernetesGetContexts();
+  const k8sClusters = await window.kubernetesGetClusters();
```

Comment:
do we know how costly it is to call these ?

Reply:
Not costly at all, as we previously used this in the render. 

I did make them into a "combined" window call but noticed no difference in speed.

Reply:
my question is that we're calling kubernetes API so, it may take time to grab these informations if clusters are remote (just a thought, maybe it's not true)

Reply:
I've updated this in https://github.com/containers/podman-desktop/pull/4921 and refactored it so it's less costly (we now do the conversion in the plugin rather than the renderer), since we need to get the current kube context name anyways.